cache.go

```diff
@@ -161,39 +174,65 @@ func (c *Cache) Delete(key string) bool {
 	return false
 }
 
-// Clears the cache
+// Clears the cache.
+// This is a control command.
 func (c *Cache) Clear() {
 	done := make(chan struct{})
 	c.control <- clear{done: done}
 	<-done
 }
 
-// Stops the background worker. Operations performed on the cache after Stop
-// is called are likely to panic
+// Stops the background worker.
+// This is a control command.
 func (c *Cache) Stop() {
-	close(c.promotables)
-	<-c.control
+	go func() {
+		time.Sleep(c.stopDelay)
+		select {
+		case c.control <- stop{}:
+		default:
+		}
+	}()
 }
 
 // Gets the number of items removed from the cache due to memory pressure since
-// the last time GetDropped was called
+// the last time GetDropped was called.
+// This is a control command.
 func (c *Cache) GetDropped() int {
 	res := make(chan int)
-	c.control <- getDropped{res: res}
-	return <-res
+	select {
+	case c.control <- getDropped{res: res}:
+		return <-res
+	default:
+		return 0
+	}
 }
 
 // Sets a new max size. That can result in a GC being run if the new maxium size
 // is smaller than the cached size
+// This is a control command.
 func (c *Cache) SetMaxSize(size int64) {
-	c.control <- setMaxSize{size}
+	done := make(chan struct{})
+	c.control <- setMaxSize{size: size, done: done}
```

Comment:
Is it intentional that this one is still using a blocking send whereas the others have been changed to use a non-blocking send?